test/services/WelshUIBuilderSpec.scala

```diff
@@ -0,0 +1,1388 @@
+
+      uiPage.components.length shouldBe 1
+
+      uiPage.components.head match {
+        case q: models.ui.Question =>
+          q.answers.length shouldBe 3
+          q.body.length shouldBe 2
+          q.answers.head shouldBe models.ui.Answer(Text(ans1(0)), None)
+          q.answers(1) shouldBe models.ui.Answer(Text(ans2(0)), None)
+          q.answers(2) shouldBe models.ui.Answer(Text(ans3(0)), None)
+        case _ => fail("Found non question UIComponent")
+      }
+    }
+
+    "convert a question page including answer hints into a Seq of a single Question UI object" in new Test {
+      val uiPage = uiBuilder.buildPage(questionPageWithHints.url, questionPageWithHints.stanzas.collect{case s: VisualStanza => s})
+
+      uiPage.components.length shouldBe 1
+
+      uiPage.components.head match {
+        case q: models.ui.Question =>
+          q.answers.length shouldBe 3
+          q.body.length shouldBe 2
+          q.answers.head shouldBe models.ui.Answer(Text(ans1(0)), Some(hint1))
+          q.answers(1) shouldBe models.ui.Answer(Text(ans2(0)), Some(hint2))
+          q.answers(2) shouldBe models.ui.Answer(Text(ans3(0)), Some(hint3))
+        case _ => fail("Found non question UIComponent")
+      }
+    }
+
+    "Process page with a simple instruction group" in new Test {
+      val phrase1: Phrase = Phrase(Vector("My favourite sweets are wine gums", "Fy hoff losin yw deintgig gwin"))
+      val phrase2: Phrase = Phrase(Vector("My favourite sweets are humbugs", "Fy hoff losin yw humbugs"))
+      val instruction1: Instruction = Instruction(phrase1, Seq("2"), None, true)
+      val instruction2: Instruction = Instruction(phrase2, Seq("end"), None, false)
+      val instructionGroup: InstructionGroup = InstructionGroup(Seq(instruction1, instruction2))
+      val bulletPointListStanzas = Seq(
+        KeyedStanza("start", PageStanza("/blah", Seq("1"), false)),
+        KeyedStanza("1", instructionGroup)
+      )
+      val bulletPointListPage = Page(Process.StartStanzaId, "/blah", bulletPointListStanzas, Seq.empty)
+      val uiPage = uiBuilder.buildPage(bulletPointListPage.url, bulletPointListPage.stanzas.collect{case s: VisualStanza => s})
+
+      uiPage.components.length shouldBe 1
+
+      // Check contents of bullet point list
+      val leadingTextItems: Text = Text(Words("My favourite sweets are"))
+      val bulletPointOne: Text = Text("wine gums")
+      val bulletPointTwo: Text = Text("humbugs")
+
+      uiPage.components.head match {
+        case b: BulletPointList =>
+
+          b.text shouldBe leadingTextItems
+
+          b.listItems.size shouldBe 2
+
+          b.listItems.head shouldBe bulletPointOne
+          b.listItems.last shouldBe bulletPointTwo
+        case _ => fail("Did not find bullet point list")
+      }
+    }
+
+    "Process page with a simple instruction group from prototypeJson" in new Test {
```

Comment:
These tests should fail when the correct language is used.

Reply:
pass now